lib/api/CAnomalyJobConfig.cc

```diff
@@ -520,6 +633,290 @@ void CAnomalyJobConfig::CAnalysisConfig::CDetectorConfig::parse(
+               m_FunctionName == FUNCTION_LOW_NON_NULL_SUM_ABBREV) {
+        m_Function = model::function_t::E_IndividualMetricLowNonNullSum;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_HIGH_NON_NULL_SUM ||
+               m_FunctionName == FUNCTION_HIGH_NON_NULL_SUM_ABBREV) {
+        m_Function = model::function_t::E_IndividualMetricHighNonNullSum;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_TIME_OF_DAY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationTimeOfDay
+                                  : model::function_t::E_IndividualTimeOfDay;
+        argumentRequired = false;
+        argumentInvalid = true;
+    } else if (m_FunctionName == FUNCTION_TIME_OF_WEEK) {
+        m_Function = isPopulation ? model::function_t::E_PopulationTimeOfWeek
+                                  : model::function_t::E_IndividualTimeOfWeek;
+        argumentRequired = false;
+        argumentInvalid = true;
+    } else if (m_FunctionName == FUNCTION_LAT_LONG) {
+        m_Function = isPopulation ? model::function_t::E_PopulationLatLong
+                                  : model::function_t::E_IndividualLatLong;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_MAX_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationMaxVelocity
+                                  : model::function_t::E_IndividualMaxVelocity;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_MIN_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationMinVelocity
+                                  : model::function_t::E_IndividualMinVelocity;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_MEAN_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationMeanVelocity
+                                  : model::function_t::E_IndividualMeanVelocity;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_SUM_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationSumVelocity
+                                  : model::function_t::E_IndividualSumVelocity;
+        argumentRequired = true;
+    } else {
+        if (haveSummaryCountField) {
+            LOG_ERROR(<< "Implicit function metric() cannot be "
+                         "used with a summary count field");
+            return false;
+        }
+
+        m_Function = isPopulation ? model::function_t::E_PopulationMetric
+                                  : model::function_t::E_IndividualMetric;
+
+        // This is inconsistent notation, but kept for backwards compatibility
+        m_FieldName = m_FunctionName;
+
+        return true;
+    }
+
+    // Validate
+    if (model::function_t::isPopulation(m_Function) && isPopulation == false) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() requires an 'over' field");
```

Comment:
```suggestion
        LOG_ERROR(<< "Function " << m_FunctionName << " requires an 'over' field");
```

I would ditch the brackets since they only made sense for the command line syntax.